packages/lesswrong/lib/types/schemaTypes.ts

```diff
@@ -28,24 +28,43 @@ type FieldName<N extends CollectionNameString> = (keyof ObjectsByCollectionName[
 
 type SqlFieldFunction<N extends CollectionNameString> = (fieldName: FieldName<N>) => string;
 
+type SqlJoinType = "inner" | "full" | "left" | "right";
```

Comment:
All the changes in this file and in `ProjectionContext.ts` are just type changes needed to make `isNonCollectionJoin` work.